pkg/labels/labels.go

```diff
@@ -59,10 +60,55 @@ func (ls Labels) String() string {
 		b.WriteString(strconv.Quote(l.Value))
 	}
 	b.WriteByte('}')
+	return b.String()
+}
 
+// NoRenderString returns ls as a string.
+// It uses an byte invalid character as a separator and so
+// should not be used for printing.
+func (ls Labels) NoRenderString(b *bytes.Buffer) string {
```

Comment:
You don't need two functions, return the buffer and leave the memory management to the caller.